lib/src/Uncertainty/Distribution/FisherSnedecor.cxx

```diff
@@ -142,6 +142,32 @@ NumericalScalar FisherSnedecor::computeLogPDF(const NumericalPoint & point) cons
   return normalizationFactor_ + (0.5 * d1_ - 1.0) * std::log(x) - 0.5 * (d1_ + d2_) * log1p(d1_ * x / d2_);
 }
 
+NumericalPoint FisherSnedecor::computeLogPDFGradient(const NumericalPoint & point) const
+{
+  const NumericalScalar x = point[0];
+  NumericalPoint logPdfGradient(2, 0.0);
+  if (!(x > 0.0)) return logPdfGradient;
+  const NumericalScalar d1xd2 = d1_ * x + d2_;
+  // First derivate the normlizationFactor as function of d1_, d2_ (see expression above in LogPDF)
+  // As the term is a combinations of LnBeta(d1/2, d2/2) := log(Beta(d1/2, d2/2), dLnBeta = dBeta/Beta
+  // As dBeta(x,y) = B(x,y) * (DiGamma(X) - DiGamma(x+y)) (see  https://en.wikipedia.org/wiki/Beta_function#Derivatives)
+  // it follows that d(LnBeta(x,y)) = dBeta(x,y) / Beta(x,y) = DiGamma(X) - DiGamma(x+y)
+  logPdfGradient[0] = 0.5 * ( std::log(d1_ / d2_)  + 1.0 - SpecFunc::DiGamma(0.5 * d1_) + SpecFunc::DiGamma(0.5 * d1_ + 0.5 * d2_) );
+  logPdfGradient[1] = 0.5 * (-d1_  / d2_ - SpecFunc::DiGamma(0.5 * d2_) + SpecFunc::DiGamma(0.5 * d1_ + 0.5 * d2_) );
+  // Second part is less complex
+  logPdfGradient[0] += 0.5 * (std::log(x) - log1p(d1_ * x / d2_) - (d1_ + d2_) * x / d1xd2);
+  logPdfGradient[1] += 0.5 * (-log1p(d1_ * x / d2_) + (d1_  + d2_) * (d1_ * x  / d2_) / d1xd2);
```

Comment:
I do not understand why both [0] and [1] lines are splitted.  If you factor, you have for the first term
```
std::log(d1_ / d2_)  + std::log(x) - log1p(d1_ * x / d2_) = std::log(x * d1_ /(d2_ + x * d1_))
```